app/services/vimeo/api_service.rb

```diff
@@ -1,19 +1,60 @@
 module Vimeo
   class ApiService
-    def initialize(path, current_school)
-      @path = path
+    def initialize(current_school)
       @current_school = current_school
     end
 
-    def post(data)
-      uri = URI(full_url(@path))
+    def create_video(size)
+      response = create_video_resource(size)
+      raise "Encountered error with code #{response[:error_code]} when trying to create a Vimeo video." if response[:error_code].present?
+      video_id = response[:uri].split('/')[-1]
+      add_whitelist_urls(video_id)
```

Comment:
Since this method call could involve an unbounded number of additional API calls, I think we should push this into a background job to avoid delaying the response to the UI; this should be okay since it isn't necessary for this operation to be complete _before_ the user uploads the video file.

Reply:
Hmm. I considered adding this actually, but then I thought the number of domains per school won't be that much, so throwing it into a background job might be an overkill, when a large number of schools might have only one domain? 

I can go ahead and add it if you think my assumptions above are not entirely correct.

Reply:
Pushing this into the background is just a matter of creating a job class with a `perform` method and calling `perform_later` on the class.

Given that we know that it's the _right_ thing to do, and considering how easy it is to do it, I don't think it's overkill at all. 🤔

Reply:
Hmm. Alright, I'd go ahead and add it.